hollow/src/main/java/com/netflix/hollow/tools/diff/HollowDiff.java

```diff
@@ -83,10 +101,16 @@ public HollowDiff(HollowReadStateEngine from, HollowReadStateEngine to, boolean
             schemas.addAll(toStateEngine.getSchemas());
             for (HollowSchema schema : schemas) {
                 if (schema instanceof HollowObjectSchema) {
-                    PrimaryKey pKey = ((HollowObjectSchema) schema).getPrimaryKey();
-                    if (pKey == null) continue;
+                    HollowObjectSchema objectSchema = ((HollowObjectSchema) schema);
+                    PrimaryKey pKey = objectSchema.getPrimaryKey();
+                    if (pKey==null && !isIncludeNonPrimaryKeyTypes) continue;
+
+                    // Support basic Single Field Types
+                    if (pKey==null && objectSchema.numFields()==1 && SINGLE_FIELD_SUPPORTED_TYPES.contains(objectSchema.getFieldType(0))) {
+                        pKey = new PrimaryKey(schema.getName(), objectSchema.getFieldName(0));
```

Comment:
Are the values in this field guaranteed to be unique?

Reply:
It is a single Field ObjectType so it would be interesting if Hollow allows duplicate records, it should have de-duped them.